### PR TITLE
feat(js-utils): add decorators for `throttle` and `debounce` helper functions

### PR DESCRIPTION
**Pull request by mbehzad**
== Description ==
To provide easier migration to projects using lodash-decorator to @kluntjte/js-utils, the `debounce` and `throttle` helpers are porvided as class method decorators as well.

== Closes issue(s) ==

== Changes ==

== Affected Packages ==
js-utils
services

### CODE REVIEW

File: packages/js-utils/README.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 <p>
   <img alt="Version" src="https://img.shields.io/badge/version-0.1.0-blue.svg?cacheSeconds=2592000" />
   <a href="#" target="_blank">
-    <img alt="License: (Apache)" src="https://img.shields.io/badge/License-Apache_2.0-yellow.svg" />
+    <img alt="License: ISC" src="https://img.shields.io/badge/License-ISC-yellow.svg" />
```
Comment (friewerts) [inline packages/js-utils/README.md]: this should be apache 2

File: packages/js-utils/README.md [hunk 2]
```diff
@@ -181,13 +185,11 @@ Calls API and returns JSON as Promise
 
 **Example**  
 ```js
-// use with async/await
-const myApiResponse = await fetchJSON("https://some.api/path")
+// use with async/awaitconst myApiResponse = await fetchJSON("https://some.api/path")
```
Comment (friewerts) [inline packages/js-utils/README.md]: ?

File: packages/js-utils/README.md [hunk 3]
```diff
@@ -203,11 +205,7 @@ checks, if element is in given array
...
-const fruits = ["Banana", "Orange", "Apple", "Mango"];
-
-if (hasElement(fruits, "Apple")) {
-  console.log("You got an Apple");
-}
+const fruits = ["Banana", "Orange", "Apple", "Mango"];

if (hasElement(fruits, "Apple")) {
  console.log("You got an Apple");
}
```
Comment (friewerts) [inline packages/js-utils/README.md]: ?

File: packages/js-utils/README.md [hunk 4]
```diff
@@ -244,55 +237,7 @@ merge two given arrays by the given checker function
...
//   },
//   {
//     "name": "User1",
//     "mail": "user-one@company.com"
//   },
//   {
//     "name": "User2",
//     "mail": "user-two@company.com"
//   }
// ]
```
Comment (friewerts) [inline packages/js-utils/README.md]: i like the old version

File: packages/js-utils/README.md [hunk 5]
```diff
@@ -1000,8 +882,7 @@ iterates over all nodes in a node list
 
 **Example**  
 ```js
-const buttons = document.querySelectorAll('button');
-forEachNode(buttons, (button, idx) => addClass(button, `my-button--${idx}`))
+const buttons = document.querySelectorAll('button');forEachNode(buttons, (button, idx) => addClass(button, `my-button--${idx}`))
 ```
```
Comment (friewerts) [inline packages/js-utils/README.md]: in general the new version seems a little bit buggy

File: packages/services/src/MediaQueryService.ts [hunk 6]
```diff
@@ -1,4 +1,4 @@
-import { throttle } from '@kluntje/js-utils/lib/function-helpers';
+import { throttle } from '@kluntje/js-utils/lib/function-helpers/lib/decorators';
```
Comment (friewerts) [inline packages/services/src/MediaQueryService.ts]: i think `'@kluntje/js-utils/lib/function-helpers/decorators';` would be better

File: packages/js-utils/README.md (hunk 1 above)
Comment (mbehzad) [reply packages/js-utils/README.md]: this change and other changes to the readme came after running the jsdoc2md node script.

Comment (friewerts) [reply packages/js-utils/README.md]: windows issue maybe?

Comment (mbehzad) [reply packages/js-utils/README.md]: most of them seem to be line-breakes being replaced by some characters.
Could you please run the script on the branch on your machine?

File: packages/services/src/MediaQueryService.ts [hunk 7]
```diff
@@ -1,4 +1,4 @@
-import { throttle } from '@kluntje/js-utils/lib/function-helpers';
+import { throttle } from '@kluntje/js-utils/lib/function-helpers/decorators';
```
Comment (friewerts) [inline packages/services/src/MediaQueryService.ts]: i think we need an extra build entry to achieve this import path?

**Pull request merged**